src/sankey.js

```diff
@@ -65,15 +80,35 @@ export default function Sankey() {
 
   function sankey() {
     const graph = {nodes: nodes.apply(null, arguments), links: links.apply(null, arguments)};
+    transformExtents();
     computeNodeLinks(graph);
     computeNodeValues(graph);
     computeNodeDepths(graph);
     computeNodeHeights(graph);
     computeNodeBreadths(graph);
     computeLinkBreadths(graph);
+    transformNodes(graph);
     return graph;
   }
 
+  function transformExtents() {
+    const transformedExtents = transform(x0, y0, x1, y1);
+    x0 = transformedExtents.x0;
+    y0 = transformedExtents.y0;
+    x1 = transformedExtents.x1;
+    y1 = transformedExtents.y1;
+  }
```

Comment:
This assignment pattern in `transformExtents` (and similarly in `transformNodes` below) seems a little verbose.

Without being able to reference `this` as a class-bound variable for the extents case, it's tricky to extract a common `transform(object)` interface that can handle both the extent-transformation and node-transformation cases.

If we could though, things could end up a lot cleaner:

```js
  function transformExtents() {
    transform(this);
  }

  function transformNodes({nodes}) {
    for (const node of nodes) {
      transform(node);
    }
  }
```